### ISSUE TITLE
Add support for Cloudflare

### ISSUE THREAD

**Issue by SilverBut** [troglobit/inadyn#208]
Cloudflare is a DNS provider. Though it is not meant to be a DDNS service provider, it [does](https://www.cloudflare.com/technical-resources/) support DDNS as its working method. Though `ddclient` had supported Cloudflare, the latest v4 API is not included, and now seems `ddclient` is out of maintenance.

It would be great if `inadyn` could support Cloudflare v4 API.

**Comment by troglobit** [comment]
Like most open source projects we rely on user contributions.  Hopefully someone else has the same itch and scratches it ...

**Comment by SilverBut** [comment]
And I am the one trying to scratch it ;)
Let me try. I will open a PR if possible.

**Comment by SilverBut** [comment]
@troglobit Seems not so easy. API for Cloudflare works like this:
```
+-------+                                         +------------+
|       |                                         |            |
| Agent |                                         | Cloudflare |
|       |                                         |            |
+---+---+                                         +------+-----+
    |                                                    |
    |                                                    |
    | ----list_zone()----------------------------------> |
    |                                                    |
    | <---zones<name,key>[]----------------------------- |
    |                                                    |
    | ----list_records(zone_key)-----------------------> |
    |                                                    |
    | <---records<...>[]-------------------------------- |
    |                                                    |
    |     (if exists)update_record(record_key,data)      |
    | ----or, if not exists:---------------------------> |
    |     add_record(data)                               |
    |                                                    |
    | <---result---------------------------------------- |
    |                                                    |
    |                                                    |
    |                                                    |
    |                                                    |
    +                                                    +
```

So it would need 3 RTT, which seems not supported by the plugin framework? I have checked other plugins, and they only do 1 requests per update.

**Comment by ysc3839** [comment]
See this?
https://github.com/troglobit/inadyn/blob/master/plugins/cloudxns.c

**Comment by SilverBut** [comment]
@ysc3839 oh...let me try. I had thought it was not allowed for plugins to start another HTTP transaction.

**Comment by RMerl** [comment]
Cloudflare documentation, includes one curl-based example:

https://api.cloudflare.com/#dns-records-for-a-zone-update-dns-record

**Comment by troglobit** [comment]
Support for Cloudflare added by @SimonPilkington in #253. Will be available in the upcoming v2.6 release.
